### PR TITLE
Start conversion to bicep

### PR DESCRIPTION
**Pull request by ckittel**
- Add region to the common resource group's name
- Convert common resource group deployment from JSON to bicep
- Updated API versions
- Enabled availability zones on ACR (not sure why that was disabled)
- Fix cleanup order (shared resource group was in the wrong location)

No other changes.  Ran a resource export and did a diff to ensure same results (other than names and what I called out above)

### CODE REVIEW

File: shared-svcs-stamp.bicep [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+
+var subRgUniqueString = uniqueString('aks', subscription().subscriptionId, resourceGroup().id)
+var frontDoorName = 'bicycle${subRgUniqueString}'
+
+/*** EXISTING RESOURCES ***/
+
+/*** RESOURCES ***/
+
+@description('Azure Firewall policies used in each region.')
+resource fwPoliciesBase 'Microsoft.Network/firewallPolicies@2023-09-01' = {
```
Comment (skabou) [inline shared-svcs-stamp.bicep:31]:
```suggestion
resource fwPoliciesBase 'Microsoft.Network/firewallPolicies@2023-04-01' = {
```

Comment (skabou) [reply shared-svcs-stamp.bicep:31]: I'm not seeing a newer version than 2023-04-01

File: shared-svcs-stamp.bicep [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+          name: 'org-wide-allowed'
+          priority: 100
+        }
+      ]
+    }
+  }
+}
+
+@description('Common logging sink for all resources.')
+resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2023-09-01' = {
```
Comment (skabou) [inline shared-svcs-stamp.bicep:89]:
```suggestion
resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2022-10-01' = {
```

File: shared-svcs-stamp.bicep [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+  plan: {
+    name: 'KeyVaultAnalytics(${logAnalyticsWorkspace.name})'
+    product: 'OMSGallery/KeyVaultAnalytics'
+    promotionCode: ''
+    publisher: 'Microsoft'
+  }
+}
+
+@description('Container Registry used by all clusters in this solution. Replicated to a second region.')
+resource commonAcr 'Microsoft.ContainerRegistry/registries@2023-11-01-preview' = {
```
Comment (skabou) [inline shared-svcs-stamp.bicep:164]:
```suggestion
resource commonAcr 'Microsoft.ContainerRegistry/registries@2023-01-01-preview' = {
```

Comment (ckittel) [reply shared-svcs-stamp.bicep:164]: Why are we going older?

File: shared-svcs-stamp.bicep (hunk 2 above)
Comment (ckittel) [reply shared-svcs-stamp.bicep:89]: Why are we going older?

File: shared-svcs-stamp.bicep (hunk 1 above)
Comment (ckittel) [reply shared-svcs-stamp.bicep:31]:
```output
ckittel@chkittel-desk-0:~$ az provider show --namespace Microsoft.Network --query "resourceTypes[?resourceType=='firewallPolicies'].apiVersions | [0]" --out table
Result
------------------
2023-11-01
2023-09-01
2023-06-01
2023-05-01
2023-04-01
2023-02-01
2023-01-01-preview
```

**Review by skabou — APPROVED** [review]

**Comment by skabou** [comment]
Looks good.  Apparently was referencing some outdated information.

**Comment by ckittel** [comment]
> Apparently was referencing some outdated information.

Nothing more current than asking the resource provider itself :)

Thanks for the review!

**Pull request merged**